src/ngraph/runtime/reference/quantized_convolution.hpp

```diff
@@ -0,0 +1,240 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cmath>
+
+#include "ngraph/axis_vector.hpp"
+#include "ngraph/coordinate_transform.hpp"
+#include "ngraph/runtime/reference/reverse.hpp"
+#include "ngraph/util.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            // in: NC_I...
+            // filter: C_OC_I...
+            // out: NC_O...
+            template <class INPUT, class FILTER, class OUTPUT>
```

Comment:
Scott can decide but looks like this can easily fit into the convolution.hpp by modifying the general_convolution there.

Reply:
For u8u8u8, we can probably use the existing conv ref kernel with some minimal changes, but in case of different input/filter/output type, we might need some additional changes to the definition of original ref kernel. but yes certainly doable, but this just seemed straightforward given we had a deadline to chase. will wait on @diyessi to make the call